pkg/neg/syncers/utils.go

```diff
@@ -568,7 +568,7 @@ func validatePod(pod *apiv1.Pod, nodeLister, serviceLister cache.Indexer, networ
 		count[negtypes.OtherError]++
 		return count, negtypes.ErrEPNodeTypeAssertionFailed
 	}
-	if err = nodeContainsPodIP(node, networkEndpoint); err != nil {
+	if err = nodeContainsPodIP(node, networkEndpoint); !pod.Spec.HostNetwork && err != nil {
```

Comment:
I don't know how the golang compiler translate this, but is also easier to understand if you only execute `nodeContainsPodIP(node, networkEndpoint)` if `!pod.Spec.HostNetwork`

```suggestion
        if  !pod.Spec.HostNetwork {
            if err = nodeContainsPodIP(node, networkEndpoint); && err != nil {
        }
```

Reply:
Seems + reads better

Reply:
Yes I agree. This makes the check easier to read. Updated.